staff/adri-campo/precourse/calculator/calculator.js

```diff
@@ -0,0 +1,129 @@
+/* 
+
+CALCULATOR PROJECT  
+Step by step:
+
+1) 3 decimals maximum, (.tofixed) it's used to limit decimals 
+2) The operation should have 2 parameters
+3) If it only have 1 parameter, we should give the square root (.sqrt) 
+4) Start from 0 the calculator (empty space) and save user input in the calculator
+5) All the operations and cases we can have
+6) Continue calculating or not? Question for the user, decition to take
+7) If YES, re-execute all the functions to start again
+
+*/
+
+// 1)
+
+function num(x) {
+    if (!Number.isInteger(x)) {
+    return Number.parseFloat(x.toFixed(3));
+  }
+  else {
+    return x;
+  }
+}
+
+
+  // 2)
+  
+  const add = function(num1, num2) {
+    let operation = num1 + num2;
+    return num(operation);
+  };
+  const sub = function(num1, num2) {
+    let operation = num1 - num2;
+    return num(operation);
+  };
+  const mul = function(num1, num2) {
+    let operation = num1 * num2;
+    return num(operation);
+  };
+  const div = function(num1, num2) {
+    let operation = num1 / num2;
+    if(num2 === 0) {
+      return "ERROR, that action is not possible to be executed"
+    } else {
+      return num(operation);
+    }
+  };
+
+
+// 3)
+
+const sqrt = function(num1) {
```

Comment:
When only introduces one number, instead of calculate the square, an error is thrown. This is because the function "decimals" doesn't exist, was you pretending to call function "num" from line 18 instead??

VM7780:40 Uncaught ReferenceError: decimals is not defined
    at sqrt (<anonymous>:40:5)
    at operations (<anonymous>:66:17)
    at calculator (<anonymous>:106:1)
    at <anonymous>:111:1

Reply:
You are right, as always Diana! ;) That's because in my last version I changed the name of that one and I forgot to change it here. I was crazy looking for the error because in my local file it is updated with the correct name 😫